analysis/pt.lean

```diff
@@ -0,0 +1,278 @@
+    rw measure_union _ _ _,
+    apply disjoint_iff.2 (@compl_inter_self _ a),
+    apply is_measurable.compl, repeat{assumption},
+end
+
+lemma prob_union {s : samp_space} [measurable_space s] (s₁ s₂ : set s)
+(hd : disjoint s₁ s₂) (h₁ : is_measurable s₁) (h₂ : is_measurable s₂) (p : prob s) :
+  p (s₁ ∪ s₂) = p s₁ + p s₂ := measure_theory.measure_union hd h₁ h₂ 
+
+lemma prob_diff_inter {s : samp_space} [measurable_space s](p : prob s) 
+(a b : set s) (h₁ : is_measurable a) (h₂ : is_measurable b) :
+p(b ∩ -a) + p(b ∩ a) = p(b) :=
+begin
+    have h :p(b) = p(b ∩ univ),
+    by rewrite inter_univ b,
+    rewrite [h,← compl_union_self a,set.inter_distrib_left,prob_union],
+    have g : (b ∩ -a) ∩ (b ∩ a) = ∅, by rw [inter_left_comm,set.inter_assoc, compl_inter_self,inter_empty,inter_empty],
+    apply disjoint_iff.2 g,
+    {
+        rewrite ← diff_eq,
+        apply is_measurable.diff h₂ h₁,
+    },
+    apply is_measurable.inter h₂ h₁,
+end
+
+lemma prob_union_inter {s : samp_space} [measurable_space s] (p : prob s) 
+(a b : set s) (h_1 : is_measurable a) (h_2 : is_measurable b) :
+p(a ∪ b) + p(a ∩ b) = p(a) + p(b) :=
+begin 
+    have h₁ : a ∪ b = a ∪ (b ∩ -a),by
+        rewrite [set.union_distrib_left, union_compl_self a,inter_univ],
+    have h₂ : is_measurable(b ∩ -a),
+    {
+        rewrite ← diff_eq,
+        apply is_measurable.diff h_2 h_1,
+    },
+    rewrite [h₁,prob_union a _ _ h_1 h₂],
+    have h₃: a ∩ b = b ∩ a, by exact inter_comm a b,
+    rewrite [h₃,← prob_diff_inter p a b h_1 h_2],
+    simp,
+    have h₄ : a ∩ (b ∩ -a) = ∅, by tidy,
+    apply disjoint_iff.2 h₄,
+end
+
+lemma bonferroni {s : samp_space} [measurable_space s] (p : prob s)
+(a b : set s) (h_1 : is_measurable a) (h_2 : is_measurable b) :
+p(a) + p(b) ≤ p(a ∩ b) + 1 :=
+begin
+    rewrite [← prob_union_inter p a b h_1 h_2, ← add_comm],
+    apply add_le_add_left' _,
+    apply prob_le_1,
+end
+
+lemma prob_partn {s : samp_space} [measurable_space s] (p : prob s)
+(f : ℕ → events s) 
+(h₁: is_partition' f) 
+(a : events s) 
+(h₂ : is_measurable a) 
+(h₃ : ∀ i : ℕ, is_measurable (f i))
+(i : ℕ) :
+p(a) = ∑i, p(a ∩ f i) :=
+begin
+        have g₁ : p(a) = p(a ∩ univ), by rw inter_univ,  
+        have g₂ : p(a) = p(⋃ i , (a ∩ f i)),by rw [g₁,← h₁.right, inter_Union_left],
+    rw g₂,
+    apply p.m_Union, 
+    {
+        intros j, 
+        apply is_measurable.inter, 
+        exact h₂, exact h₃ j,
+    },
+    intros m n h,
+        have g₃ : (λ (i : ℕ), a ∩ f i) m ∩ (λ (i : ℕ), a ∩ f i) n = ∅,
+        {
+        rw [set.inter_left_comm,set.inter_assoc,h₁.left m n],
+        simpa,
+        },
+    apply disjoint_iff.2 g₃,
+end
+
+end axfndns
+
+section cond_prob
+
+noncomputable def cond_prob {s : samp_space} [measurable_space s] (p : prob s)
+(a b : events s) := p(a ∩ b)/p(b)
+
+variables a b : ennreal
+
+notation `ℙ^`:95 p `[[`:95 a ` | `:95 b `]]`:0 := cond_prob p a b
+
+
+lemma cond_prob_rw {s : samp_space} [measurable_space s] (p : prob s)
```

Comment:
With the proof of this lemma, I feel it would help if `p(a)` was a `nnreal` instead of an `ennreal`. Because then you don't have to worry about the `top : ennreal` case.

Reply:
Again, doing this would come at the cost of reproving the measure theory results for probability measures, instead of being able to use them directly...